src/NuGet.Core/NuGet.Protocol/Converters/PackageDependencyGroupConverter.cs

```diff
@@ -19,9 +18,46 @@ public class PackageDependencyGroupConverter : JsonConverter
 
         public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
         {
-            var set = JsonUtility.LoadJson(reader);
+            string fxName = null;
+            List<Packaging.Core.PackageDependency> packages = new List<Packaging.Core.PackageDependency>();
+            while (reader.Read() && reader.TokenType != JsonToken.EndObject)
+            {
+                if (reader.TokenType.Equals(JsonToken.PropertyName))
+                {
+                    if (reader.Value.Equals(JsonProperties.Dependencies))
+                    {
+                        // Dependencies are stored in an array
+                        while (reader.Read() && reader.TokenType != JsonToken.EndArray)
+                        {
+                            string id = null;
+                            string version = null;
+                            while (reader.Read() && reader.TokenType != JsonToken.EndObject)
+                            {
+                                if (reader.TokenType.Equals(JsonToken.PropertyName) && reader.Value.Equals(JsonProperties.PackageId))
+                                {
+                                    reader.Read();
+                                    id = reader.Value.ToString();
```

Comment:
`reader` has a `ReadAsString()` method.

I think this code has risk of a NullReferenceException is a poorly implemented server sends json `{'id':null,}`.